pkg/apis/seedmanagement/validation/managedseed.go

```diff
@@ -0,0 +1,327 @@
+	}
+
+	// Ensure bootstrap is not changed
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newGardenlet.Bootstrap, oldGardenlet.Bootstrap, fldPath.Child("bootstrap"))...)
+
+	// Ensure merge with parent is not changed
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newGardenlet.MergeWithParent, oldGardenlet.MergeWithParent, fldPath.Child("mergeWithParent"))...)
+
+	return allErrs
+}
+
+func validateGardenletDeployment(deployment *seedmanagement.GardenletDeployment, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if deployment.ReplicaCount != nil {
+		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*deployment.ReplicaCount), fldPath.Child("replicaCount"))...)
+	}
+	if deployment.RevisionHistoryLimit != nil {
+		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*deployment.RevisionHistoryLimit), fldPath.Child("revisionHistoryLimit"))...)
+	}
+
+	if deployment.Image != nil {
+		allErrs = append(allErrs, validateImage(deployment.Image, fldPath.Child("image"))...)
+	}
+
+	allErrs = append(allErrs, metav1validation.ValidateLabels(deployment.PodLabels, fldPath.Child("podLabels"))...)
+	allErrs = append(allErrs, apivalidation.ValidateAnnotations(deployment.PodAnnotations, fldPath.Child("podAnnotations"))...)
+
+	return allErrs
+}
+
+func validateImage(image *seedmanagement.Image, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if image.PullPolicy != nil {
+		validValues := []string{string(corev1.PullAlways), string(corev1.PullIfNotPresent), string(corev1.PullNever)}
+		if !utils.ValueExists(string(*image.PullPolicy), validValues) {
+			allErrs = append(allErrs, field.NotSupported(fldPath.Child("pullPolicy"), *image.PullPolicy, validValues))
+		}
+	}
+
+	return allErrs
+}
+
+func validateGardenletConfiguration(gardenletConfig *config.GardenletConfiguration, bootstrap seedmanagement.Bootstrap, mergeWithParent bool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Ensure seed selector is not specified
+	if gardenletConfig.SeedSelector != nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("seedSelector"), "seed selector is forbidden"))
+	}
+
+	// Validate gardenlet config
+	allErrs = append(allErrs, configvalidation.ValidateGardenletConfiguration(gardenletConfig, fldPath)...)
+
+	if gardenletConfig.SeedConfig != nil {
+		seedConfigPath := fldPath.Child("seedConfig")
+
+		// Convert seed to internal version
+		seed, err := helper.ConvertSeed(&gardenletConfig.SeedConfig.Seed)
+		if err != nil {
+			allErrs = append(allErrs, field.InternalError(seedConfigPath, fmt.Errorf("could not convert seed config: %v", err)))
+			return allErrs
+		}
+
+		// Validate seed
+		allErrs = append(allErrs, validateSeed(seed, seedConfigPath)...)
+	}
+
+	if gardenletConfig.GardenClientConnection != nil {
```

Comment:
I think it makes sense to validate the SeedClientCOnnection for the case when the SeedClientConnection.Kubeconfig is not set and merge with parent is true.
This is the case where I would think that the gardenlet inherits the kubeconfig to the wrong cluster.

Mhm, but how can a gardenlet specify that he wants to use the in-cluster service account token for the Seed connection? Leave the kubeconfig field empty and merge with parent == false? 
By the same token, is it possible to use merge with parent == true and still use the in-cluster service account token for the Seed connection?

Reply:
What exactly do you suggest to validate? Having `seedClientConnection.kubeconfig` not set and `mergeWithParent` as true is perfectly valid, in fact this is what we expect to get most of the time. As we agreed previously, the controller unsets `seedClientConnection.kubeconfig` in the parent before merging (since we never expect it to be valid), so this case is not causing any issues. From my perspective, there is nothing to validate here.

I believe this also answers your other questions, please let me know if there is still something unclear.

Reply:
> unsets seedClientConnection.kubeconfig in the parent before merging (since we never expect it to be valid)

That is the part I was missing. Then it is not an issue. But should be documented, I already forgot it after our last discussions :)